### ISSUE TITLE
Tests seem to fail with new beautifulsoup

### ISSUE THREAD

**Issue by scarabeusiv** [django-compressor/django-compressor#998]
```
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_base.CacheBackendTestCase)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_base.CacheBackendTestCase)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_base.CompressorTestCase)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_base.CompressorTestCase)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_base.VerboseTestCase)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_base.VerboseTestCase)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_parsers.BeautifulSoupParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_parsers.BeautifulSoupParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_parsers.Html5LibParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_parsers.Html5LibParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_parsers.HtmlParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_parsers.HtmlParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_precompiler_class_used (compressor.tests.test_parsers.LxmlParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 270, in test_precompiler_class_used
[   27s]     self.assertEqual(output.text, 'OUTPUT')
[   27s] AssertionError: '' != 'OUTPUT'
[   27s] + OUTPUT
[   27s]
[   27s] ======================================================================
[   27s] FAIL: test_template_precompiler (compressor.tests.test_parsers.LxmlParserTests)
[   27s] ----------------------------------------------------------------------
[   27s] Traceback (most recent call last):
[   27s]   File "/usr/lib/python3.8/site-packages/django/test/utils.py", line 373, in inner
[   27s]     return func(*args, **kwargs)
[   27s]   File "/home/abuild/rpmbuild/BUILD/django_compressor-2.4/compressor/tests/test_base.py", line 295, in test_template_precompiler
[   27s]     self.assertEqual(output.text, 'p { border:10px solid green;}')
[   27s] AssertionError: '' != 'p { border:10px solid green;}'
[   27s] + p { border:10px solid green;}
[   27s]
[   27s] ----------------------------------------------------------------------
[   27s] Ran 390 tests in 8.063s
[   27s]
[   27s] FAILED (failures=14, skipped=11)
```

As you have hardcoded deps on older versions in test-requirements.txt it is passing in the travis.

**Comment by karyon** [comment]
Thanks! [beautifulsoup 4.9's changelog)[https://bazaar.launchpad.net/%7Eleonardr/beautifulsoup/bs4/view/head:/CHANGELOG], released in early april, contains this: `Embedded CSS and Javascript is now stored in distinct Stylesheet and Script tags, which are ignored by methods like get_text() `, which looks to me like it would match the test failures. 

I had a quick look at our production code and I don't think that is affected. I also think the tests don't need to be changed now, but I'll leave this open so we don't need to figure this out again when updating the dependencies the next time.

**Comment by dvzrv** [comment]
Hi! I'm packaging python-django-compressor 2.4 for Arch Linux and the failing tests are affecting the reproducibility of the package, as we are already on python-beautifulsoup 4.9.1.

**Comment by dvzrv** [comment]
We are currently doing rebuilds for python3.9 on Arch Linux.
This issue is still a problem and it breaks the tests on python3.8 and python3.9.
I can currently only disable the test suite, which is not great. :-/

How can this be fixed?

**Comment by karyon** [comment]
You could either honor the pinned versions of django-compressor's dependencies, or provide a PR that fixes the tests, or wait for someone else to do it. Personally I currently don't have the time to look into it myself, sorry.

**Comment by dvzrv** [comment]
I would like to mention that although #1057 possibly fixes this, the change is not included in the 2.4.1 release.

**Comment by diox** [comment]
2.4.1 was released before #1057. As the milestone field indicates, this will be included in the upcoming 3.0 release.

**Comment by dvzrv** [comment]
@diox thanks! I just tried to add this here so that this information is more visible (more of a self-reminder actually :D)

Thanks for the work on this! :)
